frontend/hooks/useRewardsHistory.ts

```diff
@@ -32,14 +34,15 @@ const CheckpointGraphResponseSchema = z.object({
     message: 'Expected epochLength to be a string',
   }),
   contractAddress: z.string({
-    message: 'Expected contractAddress to be a string',
+    message: 'Expected contractAddress to be a valid Ethereum address',
   }),
 });
-type CheckpointGraphResponse = z.infer<typeof CheckpointGraphResponseSchema>;
+const CheckpointsGraphResponseSchema = z.array(CheckpointGraphResponseSchema);
+type CheckpointResponse = z.infer<typeof CheckpointGraphResponseSchema>;
 
 const fetchRewardsQuery = gql`
   {
-    checkpoints(orderBy: epoch, orderDirection: desc) {
+    checkpoints(orderBy: epoch, orderDirection: desc, first: 1000) {
```

Comment:
will need to consider pagination here too, the graph has 1000 limit

can also reduce the checkpoints queried too..
```gql
// only checkpoints we support in frontend
checkpoints { 
  where: { 
    contractAddress_includes: 
      [ ALL_THE_STAKING_CONTRACT_ADDRESSES as string[] ] 
  }
}
```

```gql
// no empty checkpoints
{ 
  checkpoints (where: { serviceIds_not: []}) {
    serviceIds
  }
}
```

Reply:
i mean to process paginated data, as we're limited to 1000 records; we would need to check the next page if for example there are 1000 records in the response, and get the next 1000